### PR TITLE
Fix useGetManyReference loading state detection

### PR DESCRIPTION
**Pull request by djhi**

### CODE REVIEW

File: packages/ra-core/src/dataProvider/useGetManyReference.ts [hunk 1]
```diff
@@ -100,7 +100,8 @@ const useGetManyReference = (
                 defaultData
             ),
         }),
-        (state: ReduxState) => getTotal(state, relatedTo)
+        (state: ReduxState) => getTotal(state, relatedTo),
+        (data: any) => data.ids && data.ids.length > 0
```
Comment (fzaninotto) [inline packages/ra-core/src/dataProvider/useGetManyReference.ts:104]: Why >0 ? The dataProvider my return an empty list

File: packages/ra-core/src/reducer/admin/references/oneToMany.ts [hunk 2]
```diff
@@ -58,8 +58,9 @@ const oneToManyReducer: Reducer<OneToManyState> = (
 };
 
 export const getIds = (state: ReduxState, relatedTo: string) =>
-    state.admin.references.oneToMany[relatedTo] &&
-    state.admin.references.oneToMany[relatedTo].ids;
+    (state.admin.references.oneToMany[relatedTo] &&
+        state.admin.references.oneToMany[relatedTo].ids) ||
+    null;
```
Comment (fzaninotto) [inline packages/ra-core/src/reducer/admin/references/oneToMany.ts:63]: if the problem is `undefined` vs `null`, I'd rather you change the `isDataLoaded` to use `!= null` instead of `!== null` - less risk of breaking things.

File: packages/ra-core/src/controller/field/ReferenceManyFieldController.spec.tsx [hunk 3]
```diff
@@ -7,6 +7,76 @@ import renderWithRedux from '../../util/renderWithRedux';
...
+                reference="bar"
+                target="foo_id"
+                basePath=""
+            >
+                {props => <ComponentToTest {...props} />}
+            </ReferenceManyFieldController>,
+            {
+                admin: {
+                    references: {
+                        oneToMany: { ids: [] },
```
Comment (fzaninotto) [inline packages/ra-core/src/controller/field/ReferenceManyFieldController.spec.tsx:62]: same. Also, you test with a list of empty ids, you miss a test wit ha list of non-empty ids

File: packages/ra-core/src/controller/field/ReferenceManyFieldController.spec.tsx [hunk 4]
```diff
@@ -7,6 +7,76 @@ import renderWithRedux from '../../util/renderWithRedux';
...
+                reference="bar"
+                target="foo_id"
+                basePath=""
+            >
+                {props => <ComponentToTest {...props} />}
+            </ReferenceManyFieldController>,
+            {
+                admin: {
+                    references: {
+                        oneToMany: {},
```
Comment (fzaninotto) [inline packages/ra-core/src/controller/field/ReferenceManyFieldController.spec.tsx:27]: this is not the normal shape of the state. there is a key representing the reference between `reference` and `oneToMany`

**Review by fzaninotto — CHANGES_REQUESTED** [review]

**Review by fzaninotto — APPROVED** [review]
